packages/maskbook/src/extension/options-page/DashboardComponents/CollectibleList/index.tsx

```diff
@@ -112,73 +94,65 @@ export function CollectibleList(props: CollectibleListProps) {
-                            <CircularProgress size={25} />
-                        </Box>
-                    )}
-                </>
-            ) : (
-                <Box className={classes.root}>
-                    {dataSource.map((x) => (
-                        <div className={classes.card} key={x.tokenId}>
-                            <CollectibleCard token={x} provider={provider} wallet={wallet} />
-                            <div className={classes.description}>
-                                <Typography color="textSecondary" variant="body2">
-                                    {x.asset?.name ?? x.name}
-                                </Typography>
+            <Box className={classes.container}>
+                {collectiblesError || dataSource.length === 0 ? (
+                    <Box
+                        className={classes.root}
+                        sx={{
+                            flexDirection: 'column',
+                            alignItems: 'center',
+                            justifyContent: 'center',
+                            height: '100%',
+                        }}>
+                        <Typography color="textSecondary">No collectible found.</Typography>
+                        <Button
+                            sx={{
+                                marginTop: 1,
+                            }}
+                            variant="text"
+                            onClick={() => collectiblesRetry()}>
+                            Retry
+                        </Button>
+                    </Box>
+                ) : (
+                    <Box className={classes.root}>
+                        {dataSource.map((x) => (
+                            <div className={classes.card} key={x.tokenId}>
+                                <CollectibleCard token={x} provider={provider} wallet={wallet} />
+                                <div className={classes.description}>
+                                    <Typography color="textSecondary" variant="body2">
+                                        {x.asset?.name ?? x.name}
+                                    </Typography>
+                                </div>
                             </div>
-                        </div>
-                    ))}
-                </Box>
-            )}
+                        ))}
+                    </Box>
+                )}
+            </Box>
+            <TablePagination
+                count={-1}
+                component="div"
+                onPageChange={() => {}}
+                page={page}
+                rowsPerPage={30}
+                rowsPerPageOptions={[30]}
+                labelDisplayedRows={() => null}
+                ActionsComponent={() => {
+                    return (
+                        <Box display="flex">
+                            <IconButton size="small" disabled={page === 1} onClick={() => setPage((prev) => prev - 1)}>
```

Comment:
seems like you're reinverting part of the functionality of this TablePagination. Why?

Reply:
Because this api does not return the data about pagination(eg: total), so we need manual schedule paging.